core/db/migrate/20220317165036_set_promotions_with_any_policy_to_all_if_possible.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class SetPromotionsWithAnyPolicyToAllIfPossible < ActiveRecord::Migration[5.2]
+  def up
+    Spree::Promotion.where(match_policy: :any).includes(:promotion_rules).all.each do |promotion|
+      if promotion.promotion_rules.length <= 1
+        promotion.update(match_policy: :all)
+      end
```

Comment:
I shied away from that because it would mean also updating all adjustments with the relevant actions for the new promotions, and I'm not sure this feature is really, actively, used. I could try, but then the migration would become pretty gnarly and slow, and given that many people are on Heroku or similar hosters that don't allow for migrations to take a long time... Hm. I could try a rake task. What do you think?

Reply:
Also when we split up the promotions, the rake task would have to identify which rule applied positively if we want to do this properly. :thinking:

Reply:
A rake task would be great, but I'm also fine with a snippet of code to be pointed out in the changelog that people can customize and use as they want.

Reply:
I've added a rake task and made the migration raise if any promotions with "any" match policy and more than one rule is left. 

However, I don't have access to an existing database that has historic "any" promotions and associated adjustments. If there's anyone here who could test the rake task on real data, I'd be very interested. 

It's also messy.

Reply:
The rake task is there, and I think it will be robust.